picasso-compose/src/main/java/com/squareup/picasso3/compose/PicassoPainter.kt

```diff
@@ -40,14 +40,21 @@ fun Picasso.rememberPainter(
   request: (Picasso) -> RequestCreator
 ): Painter {
   return remember(key) { PicassoPainter(this, request, onError) }
+    .also { painter ->
+      DisposableEffect(painter) {
+        painter.start()
+        onDispose(painter::stop)
+      }
+    }
 }
 
 internal class PicassoPainter(
   private val picasso: Picasso,
   private val request: (Picasso) -> RequestCreator,
   private val onError: ((Exception) -> Unit)? = null
-) : Painter(), RememberObserver, DrawableTarget {
+) : Painter(), DrawableTarget {
 
+  private val stateObserver = SnapshotStateObserver(onChangeExecutor = { it() })
```

Comment:
This determines where subsequent calls to `request` will occur. Might want to make this post to the main thread if not already on it.